core/src/test/java/io/jenkins/pluginhealth/scoring/probes/SecurityScanGithubWorkflowProbeTest.java

```diff
@@ -0,0 +1,210 @@
+    }
+
+    @Test
+    void shouldBeAbleToDetectRepositoryWithNoSecurityScanConfigured() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Files.createDirectories(repo.resolve(".github/workflows"));
+        when(ctx.getScmRepository()).thenReturn(repo);
+
+        final ProbeResult result = probe.apply(plugin, ctx);
+        assertThat(result.status()).isEqualTo(ResultStatus.FAILURE);
+        assertThat(result.message()).isEqualTo("GitHub workflow security scan is not configured in the plugin");
+    }
+
+    @Test
+    void shouldBeAbleToDetectIncorrectWorkflowFileName() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Path workflowPath = Files.createDirectories(repo.resolve(".github/workflows"));
+        final Path workflowFile = Files.createFile(workflowPath.resolve("incorrect-file-name.yml"));
+
+        Files.write(workflowFile, List.of(SecurityScanGithubWorkflowProbe.SEARCH_LINE));
+        when(ctx.getScmRepository()).thenReturn(repo);
+
+        final ProbeResult result = probe.apply(plugin, ctx);
+        assertThat(result.status()).isEqualTo(ResultStatus.FAILURE);
+        assertThat(result.message()).isEqualTo("GitHub workflow security scan is not configured in the plugin");
+    }
+
+    @Test
+    void shouldFailIfJobsIsNotConfiguredInGitHubWorkflow() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Path workflowPath = Files.createDirectories(repo.resolve(".github/workflows"));
+        final Path workflowFile = Files.createFile(workflowPath.resolve("jenkins-security-scan.yaml@v2"));
```

Comment:
Not a "valid" file. Extension of the file `.yaml@v2` is not something that exists. It has to be `.yaml` or `.yml`.

```suggestion
        final Path workflowFile = Files.createFile(workflowPath.resolve("jenkins-security-scan.yaml"));
```